New blog: Getting started with Knative Eventing using Bitcoin transaction data (#2509)
knative/docs: blog/articles/get-started-knative-eventing.md

| Title  | LinkTitle | Date | Description | Type | Aliases

> | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | 

Reviewer: <!--whitespace-bot-->
Remove trailing whitespace:
```suggestion
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |
```

> | Getting started with Knative Eventing using Bitcoin transaction data  | Get Started with Knative Eventing  | 2020-05-01  | Guided walkthrough on using Knative Eventing  | Blog  | /blog/2010-05-01-getting-started-knative-evetning  |

Reviewer 2: nit: /blog/2010-05-01-getting-started-knative-evetning ->  /blog/2010-05-01-getting-started-knative-eventing

> | Title  | LinkTitle | Date | Description | Type | Aliases
> | ------------- | ------------- | ------------- | ------------- | ------------- | ------------- | 
> | Getting started with Knative Eventing using Bitcoin transaction data  | Get Started with Knative Eventing  | 2020-05-01  | Guided walkthrough on using Knative Eventing  | Blog  | /blog/2010-05-01-getting-started-knative-evetning  |

Reviewer 3: I think this needs to be in [Hugo frontmatter format](https://gohugo.io/content-management/front-matter/#front-matter-formats). I'd suggest using the `yaml` format, like so:

```suggestion
---
title: Getting started with Knaitve Eventing using Bitcoin transaction data
linkTitle: Bitcoin transaction data and Eventing
date: 2020-05-01
description: Walkthrough of implementing a Bitcoin transaction data source for Knative Eventing
type: "blog"
---
```

> I’ve been exploring [Knative Eventing](https://knative.dev/docs/eventing/), a system that enables a cloud native eventing ecosystem to be easily deployed through the use of **event producers** and **event consumers.**

Reviewer 2: maybe specify which version of Knative Eventing you have been using for this demo.

Author: @josiemundi would you be able to provide this info for us?

Reviewer 4: most of the work has been done in 0.11. I have also run it in 0.13.

Reviewer 4: Today I updated the repo to run on Knative eventing version 0.15.

> I’ve been exploring [Knative Eventing](https://knative.dev/docs/eventing/), a system that enables a cloud native eventing ecosystem to be easily deployed through the use of **event producers** and **event consumers.**

Reviewer 3: Before we get into "I", we should have a [byline](https://www.lexico.com/en/definition/byline) and probably a bio.

Author: good point @evankanderson 
I actually opened [an issue](https://github.com/knative/docs/issues/2510) for this and forgot to tag you.

> As part of this I put together a simple demo that someone can use to explore, what I think are, some of the key advantages of event-driven architectures, such as:

Reviewer 3: This sentence is a bit complex. Try removing some of the ambiguity:

```suggestion
I put together a simple demo to explore some of the key advantages of event-driven architectures, such as:
```

- Push-based messaging.

> - True decoupling of producers and consumers.

Reviewer 3: I'd avoid qualifiers like "true" -- my experience (at least) is that they tend to bring in more quibbles than sunlight.

Author: bringing in @josiemundi to weigh in on this one.

Reviewer 4: yes agree, we can remove "true"

- Apply business logic while data is in motion.
- Real time event-streams for data science — millisecond decisions e.g fraud detection.

> In this post, I’ll show you how to get an example eventing scenario up and running using some of the basic Knative eventing components; **broker, trigger, producer** and **consumer.** This demo shows streaming events in real-time, in-stream transformation and push-based front-ends in action and they can serve as a basis for you to explore further. You can use this as a basis and build out, exploring further with what is possible.

Reviewer 3: I think we should capitalize Broker and Trigger, since they are actual resources in Knative.

> In this post, I’ll show you how to get an example eventing scenario up and running using some of the basic Knative eventing components; **broker, trigger, producer** and **consumer.** This demo shows streaming events in real-time, in-stream transformation and push-based front-ends in action and they can serve as a basis for you to explore further. You can use this as a basis and build out, exploring further with what is possible.

Reviewer 3: ```suggestion
In this post, I’ll show you how to get an example eventing scenario up and running using some of the basic Knative eventing components; **broker, trigger, producer** and **consumer.** This demo shows streaming events in real-time, in-stream transformation, and push-based front-ends in action. You can use this as a basis and build out, exploring further with what is possible.
```

Remove a little duplication and use an [oxford comma](https://www.businessinsider.com/always-use-the-oxford-comma-2015-5) for clarity.

> This scenario uses bitcoin transaction events as an example of an event stream. The events will be displayed in real-time via and UI front-end and also classified into sizes by another service based on their total transaction value.

Reviewer 3: ```suggestion
This scenario uses Bitcoin transaction events as an example of an event stream. The events will be displayed in real-time via and UI front-end and also classified into sizes by another service based on their total transaction value.
```

(Bitcoin is a proper noun.)

In the diagram below, you can see a plan of what we will deploy.

![Diagram of this example Knative Eventing scenario](/blog/images/knative-eventing-scenario.png)

There is a github repository to accompany this demo [here](https://github.com/josiemundi/knative-eventing-blockchain-demo). The source code for all of the individual services is available in the github repo. If you want to use the pre-built images from Docker hub then you will only need the files in the yaml directory.

> This demo was presented at the first Knative Community Meetup and there is a video demo [here](https://www.youtube.com/watch?v=sGi_LuAaaT0).

Reviewer 3: I'd pull this up higher in the document; I don't know if we want to [embed a thumbnail player of the video](https://gohugo.io/content-management/shortcodes/#youtube) or not.

Author: added!

> ## Step 1: Create the Namespace and create a broker

Reviewer 3: ```suggestion
## Step 1: Create the Namespace and create a Broker
```

```
kubectl apply -f 001-namespace.yaml
```

> The first step is to deploy the 001-namespace.yaml, which creates a kubernetes namespace and automatically adds the label knative-eventing-injection=enabled. This creates a knative eventing **broker**.

Reviewer 3: ```suggestion
The first step is to deploy the 001-namespace.yaml, which creates a kubernetes namespace and automatically adds the label knative-eventing-injection=enabled. This creates a Knative Eventing **broker**.
```

> The **broker** is where events are sent to from an event-source or **producer.** It is a messaging channel, which by default is in-memory but can be something else (like a Kafka channel). From here they can be consumed by those services that are interested.

Reviewer 3: ```suggestion
The **Broker** is where events are sent to from an event-source or **producer.** It is a messaging channel, which by default is in-memory but can be something else (like a Kafka channel). From here they can be consumed by those services that are interested.
```

I'd be careful with terminology here -- eventing has both messaging Channels and the Broker (which may be backed by a messaging Channel). The key difference is that the Broker supports filtering events in the Channel based on attributes, while the Channel acts as durable storage (but doesn't perform any sort of content routing).

Author: bringing in @josiemundi for comments here.

Reviewer 4: Yes, this isn't quite right then as I have said it is a messaging channel. Should probably change to reflect what @evankanderson has said, that it may be backed by a messaging channel.

## Step 2: Deploy the Bitcoin event-source
```
kubectl apply -f 010-deployment.yaml
```

> Our **event-source** is an application that takes (unconfirmed bitcoin transaction) messages from the [blockchain.info websocket](https://www.blockchain.com/api/api_websocket) and then creates a new [CloudEvent.](https://cloudevents.io/) This is our producer of events and it *produces* events that other services *may or may not* be interested in subscribing to.

Reviewer 3: ```suggestion
Our **event-source** is an application that takes (unconfirmed Bitcoin transaction) messages from the [blockchain.info websocket](https://www.blockchain.com/api/api_websocket) and then creates a new [CloudEvent.](https://cloudevents.io/) This is our producer of events and it *produces* events that other services *may or may not* be interested in subscribing to.
```

> In our case we tell it where we want it to send the events, using the **sink** variable. The sink is passed in as an environmental variable in the deployment and, in this case, it is our broker address. To get the broker url, you can use the following command:

Reviewer 3: ```suggestion
In our case we tell it where we want it to send the events, using the **sink** variable. The sink is passed in as an environmental variable in the deployment and, in this case, it is our Broker address. To get the Broker url, you can use the following command:
```

```
kubectl get broker -n knative-eventing-websocket-source
```
Once you have deployed the event-source, you can verify it is running by getting the logs:
```

> kubectl — namespace knative-eventing-websocket-source logs -l app=wseventsource — tail=100 -f

Reviewer 2: `- namespace` -> `-namespace`

```

## Step 3: Subscribe event-display to the Bitcoin events
```
kubectl apply -f 040-trigger.yaml
```

> A **trigger** provides a filter, by which it is determined which events should be delivered to a given service. There are three triggers specified for this scenario. The subscribing service does not need to be deployed yet in order for you to set up a trigger.

Reviewer 3: ```suggestion
A **Trigger** provides a filter which selects events matching certain attributes to deliver to the specified service. There are three Triggers specified for this scenario. The subscribing service does not need to be deployed yet in order for you to set up a Trigger.
```

Trying to remove passive voice, also capitalize Trigger as a proper noun.

---
title: Getting started with Knaitve Eventing using Bitcoin transaction data
linkTitle: Bitcoin transaction data and Eventing
date: 2020-05-01
description: Walkthrough of implementing a Bitcoin transaction data source for Knative Eventing
type: "blog"
---

I’ve been exploring [Knative Eventing](https://knative.dev/docs/eventing/), a system that enables a cloud native eventing ecosystem to be easily deployed through the use of **event producers** and **event consumers.** Most of the work on this demo has been done in version 0.11, and I have also run it in version 0.13, and now it also works on version 0.15.

> This demo was presented at the first Knative Community Meetup, so you can also watch the video version here: 

Reviewer: <!--whitespace-bot-->
Remove trailing whitespace:
```suggestion
This demo was presented at the first Knative Community Meetup, so you can also watch the video version here:
```

This scenario uses Bitcoin transaction events as an example of an event stream. The events will be displayed in real-time via and UI front-end and also classified into sizes by another service based on their total transaction value.

In the diagram below, you can see a plan of what we will deploy.

![Diagram of this example Knative Eventing scenario](/blog/images/knative-eventing-scenario.png)

There is a github repository to accompany this demo [here](https://github.com/josiemundi/knative-eventing-blockchain-demo). The source code for all of the individual services is available in the github repo. If you want to use the pre-built images from Docker hub then you will only need the files in the yaml directory.

## Step 1: Create the Namespace and create a Broker
```
kubectl apply -f 001-namespace.yaml
```

The first step is to deploy the 001-namespace.yaml, which creates a kubernetes namespace and automatically adds the label knative-eventing-injection=enabled. This creates a Knative Eventing **broker**.

> The **Broker** is where events are sent to from an event-source or **producer.** It may be backed as a messaging channel, which by default is in-memory but can be something else (like a Kafka channel). From here they can be consumed by those services that are interested. 

Reviewer: <!--whitespace-bot-->
Remove trailing whitespace:
```suggestion
The **Broker** is where events are sent to from an event-source or **producer.** It may be backed as a messaging channel, which by default is in-memory but can be something else (like a Kafka channel). From here they can be consumed by those services that are interested.
```

> The **Broker** is where events are sent to from an event-source or **producer.** It may be backed as a messaging channel, which by default is in-memory but can be something else (like a Kafka channel). From here they can be consumed by those services that are interested. 

Reviewer: <!--whitespace-bot-->
Remove trailing whitespace:
```suggestion
The **Broker** is where events are sent to from an event-source or **producer.** It may be backed as a messaging channel, which by default is in-memory but can be something else (like a Kafka channel). From here they can be consumed by those services that are interested.
```
